packages/app/src/cli/services/context.ts

```diff
@@ -429,28 +431,36 @@ export async function fetchAppAndIdentifiers(
     app: AppInterface
     reset: boolean
     apiKey?: string
+    commandConfig: Config
   },
   token: string,
   reuseFromDev = true,
 ): Promise<[OrganizationApp, Partial<UuidOnlyIdentifiers>]> {
-  let envIdentifiers = getAppIdentifiers({app: options.app})
+  const app = options.app
+  let envIdentifiers = getAppIdentifiers({app})
   let partnersApp: OrganizationApp | undefined
 
-  if (isCurrentAppSchema(options.app.configuration)) {
-    const apiKey = options.apiKey ?? options.app.configuration.client_id
-    partnersApp = await appFromId(apiKey, token)
-  } else if (options.reset) {
+  if (options.reset) {
     envIdentifiers = {app: undefined, extensions: {}}
+    if (isCurrentAppSchema(app.configuration)) {
+      const configuration = await link({directory: app.directory, commandConfig: options.commandConfig})
+      app.configuration = configuration
+    }
+  }
```

Comment:
I think we changed the logic a bit here, as previously choosing `--reset` would shortcut all the other options until the `fetchOrCreateOrganizationApp` call. Or am I misreading this?

Reply:
You are right, but I'm clearing some stuff, so it should work ™️ . I had missed one scenario, it should be fine now.